### DISCUSSION TITLE
Proxmox Provider

### DISCUSSION THREAD

**Discussion by zaxlofful** [opentofu/opentofu#598] [Ideas]
I would love to see a Proxmox provider get ported here, then I could use it! Sadly right now everything I do is Proxmox...

**Comment by eranelbaz** [comment]
Hey @ZaxLofful 
This provider would be supported by https://github.com/opentofu/registry

**Comment by zaxlofful** [comment]
I don't see anything specific, is this a plan to just convert anything that is a current provider?

**Comment by mivalsten** [comment]
OpenTofu will work as drop in replacement for terraform binary. Initial plan was to use hashicorp's provider registry, but they sneakily changed their license to that only allowing hashicorp's terraform binary to download. OpenTofu team wrote basic registry that will work as github redirector (same as hashi one), and so it will support the same providers as terraform.

**Reply by zaxlofful** [reply]
Thanks! This is what I wanted to hear, just couldn't find it in the repos...

**Comment by kahnwong** [comment]
Can confirm that opentofu works with proxmox provider. This is a change made after `tofu init -upgrade`:

```bash
# .terraform.lock.hcl
-provider "registry.terraform.io/bpg/proxmox" {
+provider "registry.opentofu.org/bpg/proxmox" {
   version     = "0.32.0"
   constraints = "0.32.0"
   hashes = [
```

**Comment by sofiedotcafe** [comment]
On the topic of Proxmox with Tofu: what would be the main differences to using Nix and its way of deploying Proxmox? Debating between the two of them? I am fairly experienced on both!

Would Nix be an even better replacement not for Tofu but Ansible?

Something like [this](https://vincent.bernat.ch/en/blog/2022-cdktf-nixos) but with Tofu?

**Reply by ghost** [reply]
You can use CDKTF with `tofu` without changes, see [this issue](https://github.com/opentofu/opentofu/issues/601).
